src/Angor/Shared/WalletOperations.cs

```diff
@@ -48,22 +48,45 @@ public PsbtData CreatePsbtForTransaction(Transaction transaction, AccountInfo ac
 
         changeAddress = changeAddress ?? accountInfo.GetNextChangeReceiveAddress();
 
-        var utxoDataWithPaths = FindOutputsForTransaction((long)transaction.Outputs.Sum(_ => _.Value), accountInfo);
+        var utxoDataWithPaths = FindOutputsForTransaction((long)transaction.Outputs.Sum(_ => _.Value), accountInfo, spendAll);
         var coins = utxoDataWithPaths.Select(u => new Coin(uint256.Parse(u.UtxoData.outpoint.transactionId), (uint)u.UtxoData.outpoint.outputIndex, Money.Satoshis(u.UtxoData.value), Script.FromHex(u.UtxoData.scriptHex))).ToList();
 
         if (!coins.Any())
             throw new ApplicationException("No coins found to fund the transaction.");
 
-        var builder = new TransactionBuilder(network)
-            .AddCoins(coins)
-            .SetChange(BitcoinAddress.Create(changeAddress, network))
-            .ContinueToBuild(transaction) // Add the predefined outputs
-            .SendEstimatedFees(new FeeRate(Money.Satoshis(feeRate)))
-            .CoverTheRest(); // Ensure enough input value covers outputs + fee, adjusting change if needed
+        var clonedTransaction = network.CreateTransaction(transaction.ToHex());
 
-        var unsignedTx = builder.BuildTransaction(false);
+        foreach (var coin in coins)
+        {
+            if (clonedTransaction.Inputs.Any(x => x.PrevOut == coin.Outpoint))
+                continue;
+            var txin = new TxIn(coin.Outpoint, null);
+            txin.WitScript = new WitScript(Op.GetPushOp(new byte[72]), Op.GetPushOp(new byte[33])); // for total size calculation
+            clonedTransaction.AddInput(txin);
+        }
 
-        var psbt = NBitcoin.PSBT.FromTransaction(NBitcoin.Transaction.Parse(unsignedTx.ToHex(), nbitcoinNetwork), nbitcoinNetwork);
+        var totalSize = clonedTransaction.GetVirtualSize(4);
+        var totalFee = new FeeRate(Money.Satoshis(feeRate)).GetFee(totalSize);
+
+        if (spendAll)
+        {
+            var lastOutput = clonedTransaction.Outputs.Last();
+
+            if (totalFee >= lastOutput.Value)
```

Comment:
This seems like a short term solution, can the user control what the last output is or how much is the amount in there?